File: csharp/monogame/Apple.cs
using Microsoft.Xna.Framework;
using Microsoft.Xna.Framework.Graphics;

namespace Snake 
{
    public class Apple {
        private int x;
        private int y;

        public int X => x;
        public int Y => y;

        public void Move(int x, int y) {
            this.x = x;
            this.y = y;
        }

        public void Draw(SpriteBatch spriteBatch, Texture2D texture) {
            var pos = new Vector2(
                x * Constants.SPRITE_SIZE,
                y * Constants.SPRITE_SIZE);
            spriteBatch.Draw(texture, pos, Color.Red);
        }
    }
}

File: golang/ebiten/objects/apple.go
package objects

import (
	"image/color"
	"snake/constants"

	"github.com/hajimehoshi/ebiten/v2"
	"github.com/hajimehoshi/ebiten/v2/ebitenutil"
)

type Apple struct {
	Position Point
	Width    float64
	Height   float64
	Color    color.Color
}

func (a *Apple) Move(x, y int) {
	a.Position.X = x
	a.Position.Y = y
}

func (a *Apple) Draw(screen *ebiten.Image) {
	ebitenutil.DrawRect(
		screen,
		float64(a.Position.X*constants.SpriteSize),
		float64(a.Position.Y*constants.SpriteSize),
		a.Width,
		a.Height,
		a.Color)
}
